ports/stm32/boards/STM32F429DISC/mpconfigboard.h

```diff
@@ -72,3 +72,67 @@
 #define MICROPY_HW_USB_HS_IN_FS        (1)
 #define MICROPY_HW_USB_VBUS_DETECT_PIN (pin_B13)
 #define MICROPY_HW_USB_OTG_ID_PIN      (pin_B12)
+
+// SDRAM
+#define MICROPY_HW_SDRAM_SIZE  (64 / 8 * 1024 * 1024)  // 64 Mbit
+#define MICROPY_HW_SDRAM_STARTUP_TEST             (1)
+
+/* Timing configuration for 90 Mhz (11.90ns) of SD clock frequency (180Mhz/2) */
+#define MICROPY_HW_SDRAM_TIMING_TMRD        2
+#define MICROPY_HW_SDRAM_TIMING_TXSR        7
+#define MICROPY_HW_SDRAM_TIMING_TRAS        4
+#define MICROPY_HW_SDRAM_TIMING_TRC         7
+#define MICROPY_HW_SDRAM_TIMING_TWR         2
+#define MICROPY_HW_SDRAM_TIMING_TRP         2
+#define MICROPY_HW_SDRAM_TIMING_TRCD        2
+#define MICROPY_HW_SDRAM_REFRESH_RATE       64 /* ms */
+
+#define MICROPY_HW_SDRAM_CAS_LATENCY        3
+#define MICROPY_HW_SDRAM_COLUMN_BITS_NUM    8
+#define MICROPY_HW_SDRAM_ROW_BITS_NUM       12
+#define MICROPY_HW_SDRAM_MEM_BUS_WIDTH      16
+#define MICROPY_HW_SDRAM_INTERN_BANKS_NUM   4
+#define MICROPY_HW_SDRAM_CLOCK_PERIOD       2
+#define MICROPY_HW_SDRAM_RPIPE_DELAY        1
+#define MICROPY_HW_SDRAM_RBURST            (0)  
+#define MICROPY_HW_SDRAM_WRITE_PROTECTION  (0)  
+
+// These should be declared as just the pin name as written in pins.csv
+#define MICROPY_HW_FMC_SDCKE1            PB5
```

Comment:
Yeah I tried to figure out a way to leave it as (pin_PB5) but I couldn't see any way to have it all map through in the preprocessor rather than needing a runtime function and lookup table.

I did go back and forth about using the pin defined name or the cpu name, left it as defined name in case people wanted to name the fmc/smc pins in the csv (like the usb pins are named).
That's then more in keeping with the other pins in the file with their defined names, even though the format of them is different.

I can switch it back to pin cpu name if you think that'll be less confusing in future?

Reply:
> Yeah I tried to figure out a way to leave it as (pin_PB5) but I couldn't see any way to have it all map through in the preprocessor rather than needing a runtime function and lookup table.

It might be possible by the following kind of pattern:
```c
#define STATIC_AF(pin_obj, fn, type) \
    pin_obj == &pin_A0_obj && fn == TIM2 && type == CH1 ? 1 : \
    pin_obj == &pin_A0_obj && fn == USART2 && type == CTS ? 7 : \
    ... all possible pins and AFs! ...
    {} /* something that will give a compile error if no alt func found */
```
Yes that's a bit crazy but it should work, and would be generated, not written by hand.  It could be simplified by combining `fn` and `type` into one arg to the macro, like `STATIC_AF(pin_obj, fn_type)`, so call it as `STATIC_AF(pin_A0, TIM2_CH1)`.

It could also be split into a macro per function, eg:
```c
#define STATIC_AF_TIM2_CH1(pin_obj) \
    pin_obj == &pin_A0_obj ? 1 : \
    pin_obj == &pin_A5_obj ? 1 : \
    pin_obj == &pin_A15_obj ? 1 : \
    {}
```
But that might not be as flexible.

Reply:
That's not bad..... I've got no issue with a long generated header file for the compiler to parse through, I can't imagine it would slow compilation down too much.

Reply:
It might be wise to put this macro beast in a separate header, to be included only when needed.  Because at the moment mphalport.h is included in a lot of source files.

Reply:
This isn't as simple as I first thought.  

`fn == TIM2 && type == CH1` doesn't just work as `TIM2` and `CH1` are undefined. I've worked around this by joining them into one larger token and then making an enum of all these tokens so they've got something to match to.

I haven't been able to make this work in a macro: `pin_obj == &pin_A0_obj` 
it complains about trying to use `&` as an operator; the preprocessor can't do pointers.

`constexpr` would be perfect but we don't have access to that being C. I'll keep trying

Reply:
> I haven't been able to make this work in a macro: pin_obj == &pin_A0_obj
> it complains about trying to use & as an operator; the preprocessor can't do pointers.

That's true.  It would need to be something the *compiler* can optimise to a single constant.

Reply:
Turns out is wasn't as hard as I thought there for a bit, see latest commit!